src/object/schedcontext.c

```diff
@@ -102,6 +102,12 @@ static exception_t decodeSchedContext_Bind(sched_context_t *sc, extra_caps_t ext
         return EXCEPTION_SYSCALL_ERROR;
     }
 
+    if (sc->scRefillMax == 0 || !refill_ready(sc) || !refill_sufficient(sc, 0)) {
```

Comment:
I don't think this change should be merged. Servers cannot know the state of an SC when they invoke it for operations.

Reply:
We need some way to ensure that threads that are blocked on IPC are ready if they have scheduling contexts (unless timeout faulted). Otherwise, a passive server might end up running on a non-ready SC. So I think we would either need this, or would need to cancel IPC on the thread being bound. From discussions with @xurtis, I understand that we at least want to be able to reconfigure scheduling contexts on threads that are `BlockedOnReply`, so cancelling might not be an option.